### PR TITLE
PIM-5928: Export products without media

### PR DESCRIPTION
**Pull request by aRn0D**
When you export products without media we remove all media attributes from the generated file.

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | yes
| Added Behats                      | yes
| Changelog updated                 | yes
| Review and 2 GTM                  | yes
| Micro Demo to the PO (Story only) | TODO
| Migration script                  |
| Tech Doc                          |

### CODE REVIEW

File: src/Pim/Component/Connector/Processor/Normalization/ProductProcessor.php [hunk 1]
```diff
@@ -98,6 +98,15 @@ public function process($product)
                 ->get(JobInterface::WORKING_DIRECTORY_PARAMETER);
 
             $this->fetchMedia($product, $directory);
+        } else {
+            $mediaAttributes = $this->attributeRepository->findMediaAttributeCodes();
+            $productStandard['values'] = array_filter(
```
Comment (phaseinducer) [inline src/Pim/Component/Connector/Processor/Normalization/ProductProcessor.php]: What about `$productStandard['values'] = array_intersect_key($productStandard['values'], array_flip($mediaAttributes))`
?

Comment (phaseinducer) [inline src/Pim/Component/Connector/Processor/Normalization/ProductProcessor.php]: What about:
```
$productStandard['values'] = array_diff_key($productStandard['values'], array_flip($mediaAttributes))
```
?
Dunno what is the most readable :/

**Comment by damien-carcel** [comment]
Same comment than @phaseinducer , otherwise GTM

File: src/Pim/Component/Connector/Processor/Normalization/ProductProcessor.php (hunk 1 above)
Comment (aRn0D) [inline src/Pim/Component/Connector/Processor/Normalization/ProductProcessor.php]: I'd rather prefer to use `filter`, `reduce` and `map` functions to manipulate array or collection. No need to call several functions, we just need to apply a pure function on your collection.

**Comment by phaseinducer** [comment]
GTM

**Comment by SamirBoulil** [comment]
GTM

**Pull request merged**